ocs_ci/ocs/container.py

```diff
@@ -0,0 +1,315 @@
+
+import logging
+import time
+import concurrent.futures
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.exceptions import CommandFailed
+
+from tests.helpers import (create_unique_resource_name,
+                           get_admin_key, get_cephfs_name,
+                           get_cephfs_data_pool_name)
+
+log = logging.getLogger(__name__)
+
+
+class Container(object):
```

Comment:
@vasukulkarni - I see most of the operations this class tries to do already exists in a different form. when do we intend to use this class? class name and description don't seem to reflect the purpose of this class.

Reply:
@keesturam yes they do exist and as a separate object, The class here I combined for *usecases* where tests can just Init them in fixture(within the test) and all dependencies are created in order to get the pvc object,  Also the class gives cleaner picture of what's happening vs digging through various functions on how they are linked. Makes it easier to enhance/add other class methods.

eg: 
```python
@pytest.fixture(scope='class')
def myobj(request):
    # Create Secret and Pool
    secret = helpers.create_secret(constants.CEPHBLOCKPOOL)
    pool = helpers.create_ceph_block_pool()
    # Create storage class
    log.info("Creating a Storage Class")
    sc = helpers.create_storage_class(sc_name='my-sc-name',
                                      interface_type=constants.CEPHBLOCKPOOL,
                                      secret_name=secret.name,
                                      interface_name=pool.name)


    def teardown():
        sc.delete()
        secret.delete()
        pool.delete()
    request.addfinalizer(teardown)
```

vs

```python
@pytest.fixture(scope='class')
def myobj(request):
    logger.info("Creating a Storage Class")
    pvc = Container(sc_name='my-sc-name', cleanup=True)
    pvc.create()

    def teardown():
        pvc.delete()
    request.addfinalizer(teardown)

```

Reply:
sure, sounds reasonable. 
Please change the class name and description accordingly. 'Container' definitely sounds inappropriate :-)